pkg/operator/ceph/cluster/utils_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage Kubernetes storage.
+package cluster
+
+import (
+	ctx "context"
+	"testing"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/operator/test"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestConfigmapOverrideUpdate(t *testing.T) {
+	// create mocked cluster context and info
+	clientset := test.New(t, 3)
+	context := &clusterd.Context{
+		Clientset: clientset,
+	}
+	ns := "test"
+	controllerRef := &metav1.OwnerReference{UID: "test-id"}
+	ownerInfo := k8sutil.NewOwnerInfoWithOwnerRef(controllerRef, "")
+	clusterMetadata := metav1.ObjectMeta{
+		Namespace: ns,
+	}
+
+	// The configmap should be created without any labels/annotations when helm not installed
+	err := populateConfigOverrideConfigMap(context, ns, ownerInfo, clusterMetadata)
+	assert.NoError(t, err)
+	cm, err := clientset.CoreV1().ConfigMaps(ns).Get(ctx.TODO(), k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(cm.Annotations))
+	assert.Equal(t, 0, len(cm.Labels))
+
+	// Remove the labels and annotations from the configmap and verify they are added back
+	// This tests the upgrade scenario where the labels and annotations will not exist
+	cm.Labels = nil
+	cm.Annotations = nil
+	_, err = clientset.CoreV1().ConfigMaps(ns).Update(ctx.TODO(), cm, metav1.UpdateOptions{})
+	assert.NoError(t, err)
+	err = populateConfigOverrideConfigMap(context, ns, ownerInfo, clusterMetadata)
+	assert.NoError(t, err)
+	cm, err = clientset.CoreV1().ConfigMaps(ns).Get(ctx.TODO(), k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(cm.Annotations))
+	assert.Equal(t, 0, len(cm.Labels))
```

Comment:
```suggestion
	assert.Equal(t, 0, len(cm.Annotations))
	assert.Equal(t, 0, len(cm.Labels))
```

The above comment says `verify they are added back` but the len is zero for annotation and labels. am I missing something?

Reply:
I think we just need to update the comment above, I see below the same comments but len is not zero.